CONTRIBUTING.md

```diff
@@ -1,25 +1,130 @@
+- If it's a feature:
+  - Your title should be a quick summary such as "Add ability to send encrypted files"
+  - Your description should describe the outcome, not the solution.
+    - For instance: "A function exists which can be used to send encrypted files to Matrix rooms".
+    - Not: "There should be a MatrixClient.sendEncryptedFile() so the Foobar bridge can send encrypted images"
+
+Issues will be categorised according to [Synapse's rules](https://github.com/matrix-org/synapse/issues/9460). To summarise:
+
+- Your issue will be catagorised as one of the following types by the maintainer team:
+  - **T-Defect**: Bugs, crashes, hangs, vulnerabilities, or other reported problems.
+  - **T-Enhancement**: New features, changes in functionality, performance boosts, user-facing improvements.
+  - **T-Documentation**: Improvements or additions to documentation.
+- You *may* have a severity assigned as an estimate to understand how bad the problem is and how many users are affected.
+
+The assigned labels are at the maintainers' discretion, and we will make every effort to be transparent when triaging. We do
+not, as a rule, assign priority labeling to issues.
+
+## Contributing documentation
+
+Documentation is important to us, as bridges are complex beasts and rely on good documentation for both
+administrators and users. There are a couple of things to keep in mind when when writing documentation
+for bridge projects:
+ 
+ - Our documentation is currently English, though eventually we might be able to adopt a multi-lingual documentation system.
+ - Use [Plain English](https://en.wikipedia.org/wiki/Plain_English) when documenting. Assume a non-native speaker audience.
+ - Please take care to proofread.
+ - Documentation should be written for both end users of the bridge, as well as system administrators. It should always be
+   made clear in the text which class of user you are targetting.
+
+## Contributing code
+
+First of all, thank you for considering making a change to one of our projects. Every change no matter the size makes a difference! 
+
+### 🖌️ Code style
+
+Each repository contains an `eslint` configuration which will dictate the style of the code. All code should be written in TypeScript. At time of
+writing, we target ES2020 (the latest version supported by Node 14). The CI will lint your code automatically, but you can save yourself some time by running (`yarn lint`/`npm lint`) to check locally.
+
+### 🧪 Tests / CI
+
+To test your changes, you can run the `test` command with either `yarn test` or `npm test`. Some projects may have additional testing constraints
+noted in the project-specific section below.
+
+Please be aware that reviewers will expect CI to be passing before your changes will be approved, and priority will be given to PRs that
+pass CI when reviewing too. If you can't get the CI to pass, please reach out to us (and do not assume that it's always your change that broke it!).
```

Comment:
> please reach out to us

What is the best way to reach out to maintainers of each project (PR comment, message in a Matrix room, other)? Is the preferred way universal across matrix-org or is it per-project (e.g. one project will prefer PR comments, another will prefer conversations in a matrix room)?

Reply:
I think we'll probably mention this in each specific contributing file.